src/app/clusters/sample-mei-server/sample-mei-server.cpp

```diff
@@ -0,0 +1,228 @@
+
+    // registers the fixed endpoints
+    for (uint16_t endpointIndex = 0; endpointIndex < FIXED_ENDPOINT_COUNT; endpointIndex++)
+    {
+        auto endpoint = emberAfEndpointFromIndex(endpointIndex);
+        ChipLogProgress(Zcl, "Creating cluster, Ep index %d", endpoint);
+        if (emberAfContainsServer(endpoint, SampleMei::Id))
+        {
+            sampleMeiServer.RegisterEndpoint(endpoint);
+        }
+    }
+}
+
+void MatterSampleMeiClusterServerShutdownCallback(chip::EndpointId endpoint)
+{
+    sampleMeiServer.Shutdown();
+}
+
+// *****************************************************************************
+// SampleMeiContent
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace SampleMei {
+
+SampleMeiContent::SampleMeiContent() : SampleMeiContent(kInvalidEndpointId) {}
+
+SampleMeiContent::SampleMeiContent(EndpointId aEndpoint)
+{
+    endpoint = aEndpoint;
+
+    // Attribute default values
+    flipflop = false;
+}
+
+// *****************************************************************************
+// SampleMeiServer
+
+void SampleMeiServer::InvokeCommand(HandlerContext & ctxt)
+{
+    auto endpoint      = ctxt.mRequestPath.mEndpointId;
+    auto endpointIndex = EndpointIndex(endpoint);
+    if (endpointIndex == std::numeric_limits<size_t>::max())
+    {
+        ctxt.mCommandHandler.AddStatus(ctxt.mRequestPath, Protocols::InteractionModel::Status::UnsupportedEndpoint);
+        return;
+    }
+
+    switch (ctxt.mRequestPath.mCommandId)
+    {
+    case Commands::Ping::Id:
+        HandleCommand<Commands::Ping::DecodableType>(ctxt, [endpoint](HandlerContext & ctx, const auto & req) {
+            ChipLogProgress(Zcl, "Ping Command on endpoint %d", endpoint);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Success);
+        });
+        return;
+    case Commands::AddArguments::Id:
+        HandleCommand<Commands::AddArguments::DecodableType>(ctxt, [endpoint](HandlerContext & ctx, const auto & req) {
+            ChipLogProgress(Zcl, "AddArgumentsCommand on endpoint %d", endpoint);
+            if (req.arg1 > UINT8_MAX - req.arg2)
+            {
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand);
+                return;
+            }
+
+            AddArgumentsResponse::Type response;
+            response.returnValue = static_cast<uint8_t>(req.arg1 + req.arg2);
+            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+        });
+        return;
+    }
+}
+
+CHIP_ERROR SampleMeiServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err     = CHIP_NO_ERROR;
+    auto endpoint      = aPath.mEndpointId;
+    auto endpointIndex = EndpointIndex(endpoint);
+    if (endpointIndex == std::numeric_limits<size_t>::max())
+    {
+        return CHIP_ERROR_NO_ENDPOINT;
```

Comment:
This will turn into FAILURE on the wire.  Is that the intent?

Generally, CHIP_ERROR_NO_ENDPOINT is an internal error to do with UDP/TCP "endpoints".

Reply:
I see. I misunderstood the error code, changed to other errors